keyvalue/clients/src/main/java/tachyon/client/keyvalue/hadoop/KeyValueRecordWriter.java

```diff
@@ -23,34 +23,56 @@
 import org.apache.hadoop.util.Progressable;
 import org.apache.http.annotation.ThreadSafe;
 
+import tachyon.TachyonURI;
 import tachyon.client.keyvalue.KeyValueStoreWriter;
 import tachyon.client.keyvalue.KeyValueStores;
 import tachyon.exception.TachyonException;
 
 /**
- * A {@link RecordWriter} to write key-value pairs output by Reducers to a {@link KeyValueStores}.
+ * It writes key-value pairs into a temporary key-value store.
  */
 @ThreadSafe
 class KeyValueRecordWriter implements RecordWriter<BytesWritable, BytesWritable> {
-  private KeyValueStoreWriter mWriter;
-  private Progressable mProgress;
+  private static final KeyValueStores KEY_VALUE_STORES = KeyValueStores.Factory.create();
```

Comment:
KEY_VALUE_STORES looks kind of weird. Usually we would make something a non-static field. Is there a reason it needs to be static?